frontend/src/component/project/Project/ProjectSettings/ProjectActions/ProjectActionsTable/ProjectActionsModal/ProjectActionsForm/useProjectActionsForm.ts

```diff
@@ -94,7 +101,7 @@ export const useProjectActionsForm = (action?: IActionSet) => {
         return true;
     };
 
-    const validateActions = (actions: IAction[]) => {
+    const validateActions = (actions: UIAction[]) => {
         if (actions.length === 0) {
             setError(ErrorField.ACTIONS, 'At least one action is required.');
             return false;
```

Comment:
If we don't want empty actions or filters, maybe we can add validation for that, following this flow instead of the realtime check.

Reply:
Yes, I need to learn how to do that, it felt like too much for a single PR, as I may be write scrappy code (mostly because I'm not experienced in UI dev)